apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -0,0 +1,367 @@
+        }
+
+        /// <inheritdoc />
+        protected override DbParameterCollection DbParameterCollection => Parameters;
+
+        /// <inheritdoc />
+        protected override DbTransaction DbTransaction
+        {
+            get { return _transaction; }
+            set { _transaction = (SpannerTransaction) value; }
+        }
+
+        /// <summary>
+        /// </summary>
+        internal SpannerCommandTextBuilder SpannerCommandTextBuilder { get; set; }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        public object Clone()
+        {
+            return new SpannerCommand {
+                DesignTimeVisible = DesignTimeVisible,
+                Parameters = Parameters,
+                SpannerConnection = SpannerConnection,
+                SpannerCommandTextBuilder = SpannerCommandTextBuilder,
+                CommandTimeout = CommandTimeout
+            };
+        }
+
+        /// <inheritdoc />
+        public override void Cancel()
+        {
+            _synchronousCancellationTokenSource.Cancel();
+        }
+
+        /// <inheritdoc />
+        public override int ExecuteNonQuery()
+        {
+            return ExecuteNonQueryAsync(_synchronousCancellationTokenSource.Token).Result;
+        }
+
+        /// <inheritdoc />
+        public override async Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToken)
+        {
+            // There must be a valid and open connection.
+            if (SpannerConnection == null)
+                throw new InvalidOperationException(
+                    "You must assign a SpannerConnection to this command to execute it.");
+            if (SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Delete
+                && SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Insert
+                && SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.InsertOrUpdate
+                && SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Update)
+                throw new InvalidOperationException(
+                    "You can only call ExecuteReader on a Delete, Insert, InsertUpdate, or Update Command");
+            if (!SpannerConnection.IsOpen)
+                await SpannerConnection.OpenAsync(cancellationToken).ConfigureAwait(false);
+            if (!SpannerConnection.IsOpen)
+                throw new InvalidOperationException(
+                    "Unable to open the Spanner connection to the database to execute the query.");
+
+            // Execute the command.
+            var mutations = new List<Mutation>();
+            if (SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Delete)
+            {
+                var w = new Mutation.Types.Write {Table = SpannerCommandTextBuilder.TargetTable};
+                w.Columns.AddRange(Parameters.Cast<SpannerParameter>().Select(x => x.SourceColumn ?? x.ParameterName));
```

Comment:
not working for Mutation..Write.Values.  can you verify?  It has no setter and ditto for ListValue.